src/Component/PreviewMap/PreviewMap.tsx

```diff
@@ -0,0 +1,272 @@
+import * as React from 'react';
+
+import OlMap from 'ol/Map';
+import OlLayerVector from 'ol/layer/Vector';
+import OlSourceVector from 'ol/source/Vector';
+import OlFormatGeoJSON from 'ol/format/GeoJSON';
+import OlGeomPoint from 'ol/geom/Point';
+import OlGeomLineString from 'ol/geom/LineString';
+import OlGeomPolygon from 'ol/geom/Polygon';
+import OlView from 'ol/View';
+import OlFeature from 'ol/Feature';
+import OlLayerTile from 'ol/layer/Tile';
+import OlSourceOSM from 'ol/source/OSM';
+
+import { Style } from 'geostyler-style';
+
+import './PreviewMap.less';
+
+import 'ol/ol.css';
+
+import OlStyleParser from 'geostyler-openlayers-parser';
+
+const _isEqual = require('lodash/isEqual');
+
+import { Data, VectorData } from 'geostyler-data';
+
+import { localize } from '../LocaleWrapper/LocaleWrapper';
+
+// default props
+export interface PreviewMapDefaultProps {
+  projection: string;
+  dataProjection: string;
+  showOsmBackground: boolean;
+  mapHeight: number;
+}
+
+// non default props
+export interface PreviewMapProps extends Partial<PreviewMapDefaultProps> {
+  data?: Data;
+  style: Style;
+  map?: any;
+  layers?: any[];
+  controls?: any[];
+  interactions?: any[];
+  onMapDidMount?: (map: any) => void;
+}
+
+// state
+interface PreviewMapState {
+  mapTargetId: string;
+}
+
+/**
+ * Symbolizer preview UI.
+ */
+export class PreviewMap extends React.Component<PreviewMapProps, PreviewMapState> {
+
+  _styleParser = new OlStyleParser();
+
+  /** reference to the underlying OpenLayers map */
+  map: any;
+
+  /** refrence to the vector layer for the passed in features  */
+  dataLayer: any;
+
+  public static defaultProps: PreviewMapDefaultProps = {
+    projection: 'EPSG:3857',
+    dataProjection: 'EPSG:4326',
+    showOsmBackground: true,
+    mapHeight: 267
+  };
+
+  constructor(props: PreviewMapProps) {
+    super(props);
+
+    const randomId = Math.floor((1 + Math.random()) * 0x10000);
+    this.state = {
+      mapTargetId: `map_${randomId}`
```

Comment:
State seems not to be needed for `mapTargetId` since it is not set/updated anywhere.